R/SimLongitudinalGSF.R

```diff
@@ -0,0 +1,188 @@
+#' @param link_dsld (`number`)\cr the link coefficient for the derivative contribution.
+#' @param link_ttg (`number`)\cr the link coefficient for the time-to-growth contribution.
+#' @param link_identity (`number`)\cr the link coefficient for the SLD Identity contribution.
+#'
+#' @slot sigma (`numeric`)\cr See arguments.
+#' @slot mu_s (`numeric`)\cr See arguments.
+#' @slot mu_g (`numeric`)\cr See arguments.
+#' @slot mu_b (`numeric`)\cr See arguments.
+#' @slot omega_b (`numeric`)\cr See arguments.
+#' @slot omega_s (`numeric`)\cr See arguments.
+#' @slot omega_g (`numeric`)\cr See arguments.
+#' @slot a_phi (`numeric`)\cr See arguments.
+#' @slot b_phi (`numeric`)\cr See arguments.
+#' @slot link_dsld (`numeric`)\cr See arguments.
+#' @slot link_ttg (`numeric`)\cr See arguments.
+#' @slot link_identity (`numeric`)\cr See arguments.
+#' @family SimLongitudinal
+#' @name SimLongitudinalGSF-class
+#' @exportClass SimLongitudinalGSF
+.SimLongitudinalGSF <- setClass(
+    "SimLongitudinalGSF",
+    contains = "SimLongitudinal",
+    slots = c(
+        sigma = "numeric",
+        mu_s = "numeric",
+        mu_g = "numeric",
+        mu_b = "numeric",
+        a_phi = "numeric",
+        b_phi = "numeric",
+        omega_b = "numeric",
+        omega_s = "numeric",
+        omega_g = "numeric",
+        link_dsld = "numeric",
+        link_ttg = "numeric",
+        link_identity = "numeric"
+    )
+)
+
+#' @rdname SimLongitudinalGSF-class
+#' @export
+SimLongitudinalGSF <- function(
+    times = c(-100, -50, 0, 50, 100, 150, 250, 350, 450, 550) / 365,
+    sigma = 0.01,
+    mu_s = c(0.6, 0.4),
+    mu_g = c(0.25, 0.35),
+    mu_b = 60,
+    a_phi = c(4, 6),
+    b_phi = c(4, 6),
+    omega_b = 0.2,
+    omega_s = 0.2,
+    omega_g = 0.2,
+    link_dsld = 0,
+    link_ttg = 0,
+    link_identity = 0
+) {
+    .SimLongitudinalGSF(
+        times = times,
+        sigma = sigma,
+        mu_s = mu_s,
+        mu_g = mu_g,
+        mu_b = mu_b,
+        a_phi = a_phi,
+        b_phi = b_phi,
+        omega_b = omega_b,
+        omega_s = omega_s,
+        omega_g = omega_g,
+        link_dsld = link_dsld,
+        link_ttg = link_ttg,
+        link_identity = link_identity
+    )
+}
+
+
+setValidity(
+    "SimLongitudinalGSF",
+    function(object) {
+        par_lengths <- c(
+            length(object@mu_s),
+            length(object@mu_g),
+            length(object@a_phi),
+            length(object@b_phi)
+        )
+        if (length(unique(par_lengths)) != 1) {
+            return("The parameters `mu_s`, `mu_g`, `a_phi`, and `b_phi` must have the same length.")
+        }
+        if (length(object@sigma) != 1) {
+            return("The parameter `sigma` must have length 1.")
+        }
+        if (length(c(object@omega_b, object@omega_s, object@omega_g)) != 3) {
+            return("The parameters `omega_b`, `omega_s`, and `omega_g` must be length 1.")
```

Comment:
this is not very precise. You could have two of them with length 0 and one with length 3